src/nycdb/plugins.py

```diff
@@ -0,0 +1,22 @@
+from pathlib import Path
+import importlib
+from importlib.util import find_spec
+import pkgutil
+
+
+class Plugin:
+    def __init__(self, name: str):
+        self.name = name
+        self.spec = find_spec(name)
+        self.root_dir = Path(self.spec.origin).parent.resolve()
+
+    def import_submodule(self, name: str):
+        return importlib.import_module(f"{self.name}.{name}")
+
+
+def iter_plugins():
+    yield Plugin('nycdb')
```

Comment:
iIf nycdb is a "plugin" than perhaps "plugin" is the right name for this then? component?

Reply:
Hmmm, I'm concerned that `Component` might also be a bit overly generic though?  I also feel like the word `plugin` has a certain understanding within the Python community (see e.g. the [Creating and discovering plugins](https://packaging.python.org/guides/creating-and-discovering-plugins/) docs on the official Python site) that conforms to what we're doing here, so I'd prefer to keep using that word if possible.

Another option is for us to actually have nycdb come with a separate package called `nycdb_core` that represents the core "built-in" plugin but I'm not sure if/how much that would help.